hash/TBBCHT.h

```diff
@@ -33,6 +33,16 @@ namespace FastHash
             }
         }
 
+        void erase(TKey idx) {
```

Comment:
So I did this just for FIFO_FH.h file. I think this is only for the FH hash table (not the DC hash)? The class is called FashHash (for the frozen hash). IN FIFO_FH.h they use (std::unique_ptr<FastHash::FashHashAPI<TValue>>  m_fasthash;). And this is the lookup for the frozen hash. (m_hash is for dynamic hash). I do not think the function has to be declared in the parent FastHashBash.h for this to work (Unless there is another reason you want it included in there).


So they did not have an erase function for the FrozenHash (only for dynamic hash).  Not sure if there would be significant trade off in setting it to a tombstone value or erasing it. If we set it to tomb stone value we would incur an extra branch when finding something in the frozen hash.

Reply:
My understanding is that m_fasthash uses `CLHT` not `TBBCHT` which inherits from `FastHash`, so we need to add the erase function to CLHT.h:
Line 295:  `m_fasthash.reset(new FastHash::CLHT<TValue>(0, align_len));`
I was saying to declare it at parent as it felt like a good practice and I assumed that TBBCHT was being used for the dynamic hash, but looks like it might have just been used as an intermediate step in testing from rereading the paper so good either way. Correct me if I'm wrong, I don't believe it's used anywhere in the code currently, every usage appears commented out.
Btw, I wasn't able to find references to `m_hash`, by chance did you mean `m_map` is for the dynamic hash or am I missing something?

Yep, I guess just something worth noting as a tradeoff / worth testing if we have time.